com.unity.render-pipelines.universal/ShaderLibrary/SSAO.hlsl

```diff
@@ -0,0 +1,452 @@
+        half4 p1b = SAMPLE_BASEMAP(uv.xy + (delta               ));
+        half4 p2a = SAMPLE_BASEMAP(uv.xy - (delta * 2.0         ));
+        half4 p2b = SAMPLE_BASEMAP(uv.xy + (delta * 2.0         ));
+        half4 p3a = SAMPLE_BASEMAP(uv.xy - (delta * 3.2307692308));
+        half4 p3b = SAMPLE_BASEMAP(uv.xy + (delta * 3.2307692308));
+
+        #if defined(BLUR_SAMPLE_CENTER_NORMAL)
+            float3x3 camProj = (float3x3)unity_CameraProjection;
+            float2 p11_22 = float2(camProj._11, camProj._22);
+            float2 p13_31 = float2(camProj._13, camProj._23);
+            half3 n0 = SampleNormal(uv, p11_22, p13_31);
+        #else
+            half3 n0 = GetPackedNormal(p0);
+        #endif
+
+        half w0 = 0.37004405286;
+        half w1a = CompareNormal(n0, GetPackedNormal(p1a)) * 0.31718061674;
+        half w1b = CompareNormal(n0, GetPackedNormal(p1b)) * 0.31718061674;
+        half w2a = CompareNormal(n0, GetPackedNormal(p2a)) * 0.19823788546;
+        half w2b = CompareNormal(n0, GetPackedNormal(p2b)) * 0.19823788546;
+        half w3a = CompareNormal(n0, GetPackedNormal(p3a)) * 0.11453744493;
+        half w3b = CompareNormal(n0, GetPackedNormal(p3b)) * 0.11453744493;
+
+        half s;
+        s  = GetPackedAO(p0)  * w0;
+        s += GetPackedAO(p1a) * w1a;
+        s += GetPackedAO(p1b) * w1b;
+        s += GetPackedAO(p2a) * w2a;
+        s += GetPackedAO(p2b) * w2b;
+        s += GetPackedAO(p3a) * w3a;
+        s += GetPackedAO(p3b) * w3b;
+
+        s /= w0 + w1a + w1b + w2a + w2b + w3a + w3b;
+    #else
+        // Faster 5-tap Gaussian with linear sampling
+        half4 p0  = SAMPLE_BASEMAP(uv.xy                         );
+        half4 p1a = SAMPLE_BASEMAP(uv.xy - (delta * 1.3846153846));
+        half4 p1b = SAMPLE_BASEMAP(uv.xy + (delta * 1.3846153846));
+        half4 p2a = SAMPLE_BASEMAP(uv.xy - (delta * 3.2307692308));
+        half4 p2b = SAMPLE_BASEMAP(uv.xy + (delta * 3.2307692308));
+
+        #if defined(BLUR_SAMPLE_CENTER_NORMAL)
+            float3x3 camProj = (float3x3)unity_CameraProjection;
+            float2 p11_22 = float2(camProj._11, camProj._22);
+            float2 p13_31 = float2(camProj._13, camProj._23);
+            half3 n0 = SampleNormal(uv, p11_22, p13_31);
+        #else
+            half3 n0 = GetPackedNormal(p0);
+        #endif
+
+        half w0 = 0.2270270270;
+        half w1a = CompareNormal(n0, GetPackedNormal(p1a)) * 0.3162162162;
+        half w1b = CompareNormal(n0, GetPackedNormal(p1b)) * 0.3162162162;
+        half w2a = CompareNormal(n0, GetPackedNormal(p2a)) * 0.0702702703;
+        half w2b = CompareNormal(n0, GetPackedNormal(p2b)) * 0.0702702703;
+
+        half s;
+        s  = GetPackedAO(p0)  * w0;
+        s += GetPackedAO(p1a) * w1a;
+        s += GetPackedAO(p1b) * w1b;
+        s += GetPackedAO(p2a) * w2a;
+        s += GetPackedAO(p2b) * w2b;
+
+        s /= w0 + w1a + w1b + w2a + w2b;
```

Comment:
wasn't this suppose to be s /= (w0 + w1a + w1b + w2a + w2b);

Reply:
We could use s *= rcp(w0 + w1a + w1b + w2a + w2b);

rcp is an approximation on div and supposedly faster and avoid precision issues with divisions close to zero.

Reply:
Good point! Fixed.